users/purge-users.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+from __future__ import print_function
+
+import sys
+import re
+import os
+from subprocess import check_output
+import optparse
+
+import yaml
+
+def list_users():
+    users = check_output("cut -d: -f1 /etc/passwd", shell=True).split('\n')
+    return users
+
+def list_devs():
+    users = list_users()
+    devs = [user for user in users if get_authorized_keys_filepath(user)]
+    return devs
+
+def get_authorized_keys_filepath(username):
+    return os.path.exists(os.path.join('/', 'home', username, '.ssh', 'authorized_keys'))
+
+def main():
+    parser = optparse.OptionParser()
+    parser.add_option('', '--keep', dest="keep_users", action="append")
+    (options, args) = parser.parse_args()
+
+    users_to_drop = []
+    for username in list_users():
+        authorized_keys = get_authorized_keys_filepath(username)
+        if authorized_keys:
+            if username not in keep_users:
```

Comment:
Could maybe replace 29-34 with `users_to_drop = [user for user in list_devs() if user not in keep_users]`?